parser.hpp

```diff
@@ -220,18 +259,15 @@ template <class GameString>
 CONSTEXPR auto parse_game_state(GameString&& get_game_state_string)
 {
     CONSTEXPR auto board = parse_board(get_game_state_string);
-    CONSTEXPR auto score = parse_score(get_game_state_string);
-    CONSTEXPR auto moves = parse_moves(get_game_state_string);
 
-    return std::make_tuple(board, score, moves);
+    return board;
```

Comment:
In retrospective, I like structs a lot better.

Reply:
Made game state as struct in a separate commit since it affected several files (still same PR), also removed parser include from engine file since engine does not use parser